common/defaults.go

```diff
@@ -317,3 +321,18 @@ func DefaultNodeSelector() map[string]string {
 		"kubernetes.io/os": "linux",
 	}
 }
+
+// GetImagePullPolicy returns the image pull policy with proper precedence:
+// 1. Global ArgoCD imagePullPolicy
+// 2. Operator environment variable setting
+// 3. Default value (lowest priority)
+func GetImagePullPolicyOnPrecedence(globalPolicy *corev1.PullPolicy, envValue string) corev1.PullPolicy {
+	if globalPolicy != nil {
+		return *globalPolicy
+	}
+
+	if envValue != "" {
+		return corev1.PullPolicy(envValue)
+	}
+	return corev1.PullPolicy("")
+}
```

Comment:
```suggestion
```
Lets drop this and handle the pull policy in `GetImagePullPolicy` func in `controllers/argoutil/resource.go`